spec/unit/shoulda/matchers/active_model/validate_presence_of_matcher_spec.rb

```diff
@@ -39,6 +39,23 @@
     end
   end
 
+  context 'a belongs_to association with a presence validation and allow_nil' do
```

Comment:
Again, it is necessary for this to be a `belongs_to` association? I realize that may be what you added this for, but I believe this can just be a normal attribute.